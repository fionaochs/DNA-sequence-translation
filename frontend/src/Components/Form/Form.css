form {
    display: flex;
    justify-content: center;
    flex-direction: row;
    align-items: center;
    /*font-size: 1.2rem;*/
}

button {
    color: rgb(250, 248, 248);
    cursor: pointer;
    /*font-size: 1.2rem;*/
    outline: rgb(0, 149, 255);
    background-color: #1A8CA9;
    /*background-color: #1A8CA9;*/
    /*background-color: rgb(0, 157, 255);*/
    border-radius: .2rem;
    border-color: #23b4d8;
    margin: 1rem;
    padding: 0.25rem 0.3rem 0.25rem 0.3rem;
}

button:hover {
    background-color: rgb(250, 248, 248);
    cursor: pointer;
    color: #1A8CA9;
}

input {
    border-radius: .2rem;
    padding: .2rem;
}