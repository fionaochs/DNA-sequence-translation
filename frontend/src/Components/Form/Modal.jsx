import React from 'react'
import styles from './Form.css';

const SearchModal = ({DNASequence}) => {

    return (
        <div className={styles.modalDiv}>
            <h3>Starting async task to find your protein with submitted sequence <span>{DNASequence}</span></h3>
            <h3>Close modal and hit <span>Get results</span> to see your results</h3>
        </div>
    )

}

export default SearchModal;