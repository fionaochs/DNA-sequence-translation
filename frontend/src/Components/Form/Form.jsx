import React, {useState} from 'react';
import {FetchResults, PostProtein} from "../../services/requests";
import {useLocalStorage} from "../../useLocalStorage";
import styles from './Form.css';
import Modal from 'react-modal';
import SearchModal from './Modal';

const Form = () => {
        const [proteinName, setProteinName] = useState('');
        const [results, setResults] = useLocalStorage("results", []);
        const [modalIsOpen, setModalIsOpen] = useState(false);

        const handleChange = ({target}) => setProteinName(target.value);

        const setModalIsOpenToTrue = () => {
            setModalIsOpen(true)
        }
        const setModalIsOpenToFalse = () => {
            setModalIsOpen(false)
        }
        const HandleClick = (e) => {
            e.preventDefault();

            setModalIsOpenToTrue();

            PostProtein(proteinName)
                .then(res => {
                    // window.location.reload(false);
                })
        };
        const getResults = (e) => {
            e.preventDefault();

            FetchResults()
                .then(res => {
                    if (res === undefined) {
                        setResults([...results, {'id': -1, 'DNASequence': proteinName}])
                    } else {
                        setResults(res)
                    }
                    window.location.reload(false);
                })
        };
        const customStyles = {
            content: {
                top: '50%',
                left: '50%',
                right: 'auto',
                bottom: 'auto',
                marginRight: '-50%',
                transform: 'translate(-50%, -50%)',
                backgroundColor: '#1A8CA9',
                width: '30rem'
            },
            overlay: {
                background: "#1c49a4"
            }
        };
        return (
            <form onSubmit={HandleClick}>
                <input type="text" value={proteinName} onChange={handleChange}
                       label="Sequence" placeholder="Sequence"/>
                <button onClick={HandleClick}>Find protein</button>
                <button onClick={getResults}>Get results</button>

                <Modal isOpen={modalIsOpen} style={customStyles} onRequestClose={setModalIsOpenToFalse}>
                    <button onClick={setModalIsOpenToFalse}>x</button>
                    <SearchModal DNASequence={proteinName}/>
                </Modal>
            </form>
        );
    }
;

export default Form;