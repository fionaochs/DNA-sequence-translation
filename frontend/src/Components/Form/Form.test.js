import React from 'react';
import {render, cleanup} from '@testing-library/react';
import Form from './Form';

describe('Form component', () => {
    afterEach(() => cleanup());
    it('renders Form', () => {
        const {asFragment} = render(<Form/>);
        expect(asFragment()).toMatchSnapshot();
    });
    it('changes the inputted sequence value', async () => {
        render(<Form/>);
        const textInput = await screen.getByLabelText('Sequence');
screen.
        fireEvent.change(textInput, {target: {value: 'AAGTTA'}});
        await waitFor(() => {
            expect(textInput.value).toBe('AAGTTA');
        });
    });
});