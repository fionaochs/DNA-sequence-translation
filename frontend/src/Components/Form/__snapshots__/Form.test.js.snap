// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<DocumentFragment>
  <form>
    <input
      placeholder="Sequence"
      type="text"
      value=""
    />
    <button>
      Find protein
    </button>
  </form>
</DocumentFragment>
`;
